pcdsdevices/__init__.py

```diff
@@ -12,5 +13,25 @@ def __contains__(self, value):
 del Device
 del __contains__
 
+
+# Fix handling of often corrupt IMS PN fields
+def make_new_bts(old_bts):
+    def new_bts(st1):
+        try:
+            return old_bts(st1)
+        except UnicodeDecodeError:
+            return new_bts(st1[1:])
```

Comment:
Can we try instead `.decode(st1, "ignore")`? https://docs.python.org/3/howto/unicode.html

Reply:
Yes please, taking all suggestions here

Reply:
Here is the string we are dealing with btw:
```
$ caget XPP:USR:PRT:MMS:25.PN
XPP:USR:PRT:MMS:25.PN          \377\375\003\377\373\003\377\373\001MFI3CRL17N4-EE
```

Reply:
```python
In [6]: b'\377\375\003\377\373\003\377\373\001MFI3CRL17N4-EE'.decode("ascii", "ignore")
Out[6]: '\x03\x03\x01MFI3CRL17N4-EE'

In [7]: b'\377\375\003\377\373\003\377\373\001MFI3CRL17N4-EE'.decode("latin-1", "ignore")
Out[7]: 'ÿý\x03ÿû\x03ÿû\x01MFI3CRL17N4-EE'

In [8]: b'\377\375\003\377\373\003\377\373\001MFI3CRL17N4-EE'.decode("utf-8", "ignore")
Out[8]: '\x03\x03\x01MFI3CRL17N4-EE'
```

I think stripping out those bad characters then should be pushed to the component, but at least the above won't raise

Reply:
Yep, it definitely works with ignore. The other way we could approach this is to set the pyepics decoding to use the latin-1 character set via changing the environment variable:
https://github.com/pyepics/pyepics/blob/e2370d277f9a1ca200740446fefa2cac234462b9/epics/utils.py#L7